utils/efushell/SocketDriver.py

```diff
@@ -6,30 +6,32 @@
 
 
 class SimpleSocket:
-  def __init__(self, hostname = "localhost", port = 8888, timeout = 2):
-    self.access_semaphor = threading.Semaphore(1)
-    try:
-        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    except socket.error, msg:
-        sys.stderr.write("socket() [Socket connection error] Cannot connect to %s, error: %s\n" % (hostname, msg[0]))
-        sys.exit(1)
+    def __init__(self, hostname="localhost", port=8888, timeout=2):
+        self.access_semaphor = threading.Semaphore(1)
+        try:
+            self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        except socket.error:
+            sys.stderr.write(
+                "socket() [Socket connection error] Cannot connect to %s, error: %s\n" % (hostname, socket.error))
+            sys.exit(1)
 
-    self.sock.settimeout(timeout)
+        self.sock.settimeout(timeout)
 
-    try:
-        self.sock.connect((hostname, port))
-    except socket.error, msg:
-        sys.stderr.write("connect() [Socket connection error] Cannot connect to %s:%d, error: %s\n" % (hostname, port, msg[0]))
-        sys.exit(2)
+        try:
+            self.sock.connect((hostname, port))
+        except socket.error:
+            sys.stderr.write("connect() [Socket connection error] Cannot connect to %s:%d, error: %s\n" % (
+            hostname, port, socket.error))
+            sys.exit(2)
 
-  def SendCommand(self, cmd):
-    self.access_semaphor.acquire()
-    self.sock.send(cmd + '\n')
-    self.access_semaphor.release()
+    def SendCommand(self, cmd):
+        self.access_semaphor.acquire()
+        self.sock.send('{}\n'.format(cmd).encode('utf-8'))
```

Comment:
not that I object to this, but was it necessary? To me it is less readable.

Reply:
`encode` needs to be called on the full string, so I don't think I could do the string concat on the same line.
This would work:
```python
cmd += '\n'
self.sock.send(cmd.encode('utf-8'))
```

Reply:
If we were able to require Python 3.6 then we could use:
```python
self.sock.send(f'{cmd}\n'.encode('utf-8'))
```